programs/system/src/instructions/advance_nonce_account.rs

```diff
@@ -37,11 +37,12 @@ impl AdvanceNonceAccount<'_> {
             AccountMeta::readonly_signer(self.authority.key()),
         ];
 
-        // instruction
+        // instruction data uses 4-byte LE discriminator expected by the system program
+        const DATA: [u8; 4] = [4, 0, 0, 0];
         let instruction = Instruction {
             program_id: &crate::ID,
             accounts: &account_metas,
-            data: &[4],
+            data: &DATA,
```

Comment:
maybe this way will be simpler?
```suggestion
            data: &[4, 0, 0, 0],
```